src/test/regress/expected/with_executors.out

```diff
@@ -0,0 +1,386 @@
+	),
+	join_last_two AS (
+		SELECT 
+			router_exec.user_id, local_table.id 
+		FROM 
+			router_exec 
+		JOIN 
+			local_table 
+		ON 
+			router_exec.user_id=local_table.id
+	)
+	SELECT * FROM join_first_two JOIN join_last_two ON id = value_3 ORDER BY 1,2,3,4,5 LIMIT 10
+)
+SELECT DISTINCT uid_1, time, value_3 FROM cte ORDER BY 1, 2, 3 LIMIT 20;
+ uid_1 |              time               | value_3 
+-------+---------------------------------+---------
+     2 | Wed Nov 22 18:19:49.944985 2017 |       1
+(1 row)
+
+-- All combined with outer join
+WITH cte AS (
+	WITH task_tracker AS (
+		SELECT 
+			users_table.user_id as uid_1, users_table.value_2 as val_2
+		FROM 
+			users_table 
+		JOIN
+			events_table 
+		ON 
+			users_table.value_2=events_table.value_2
+	),
+	real_time AS (
+		SELECT * FROM users_table
+	),
+	router_exec AS (
+		SELECT * FROM events_table WHERE user_id = 1
+	),
+	local_table AS (
+		SELECT * FROM local_table
+	),
+	join_first_two AS (
+		SELECT uid_1, time, value_3 FROM task_tracker JOIN real_time ON val_2=value_3
+	),
+	join_last_two AS (
+		SELECT 
+			router_exec.user_id, local_table.id 
+		FROM 
+			router_exec 
+		JOIN 
+			local_table 
+		ON 
+			router_exec.user_id=local_table.id
+	)
+	SELECT uid_1, value_3 as val_3 FROM join_first_two JOIN join_last_two ON id = value_3 ORDER BY 1,2 LIMIT 10
+)
+SELECT DISTINCT uid_1, val_3 FROM cte join events_table on cte.val_3=events_table.event_type ORDER BY 1, 2;
+ uid_1 | val_3 
+-------+-------
+     2 |     1
+(1 row)
+
+-- CTEs should not be able to terminate (the last SELECT) in a local query
+WITH cte AS (
+	SELECT * FROM users_table
+)
+SELECT count(*) FROM cte JOIN local_table ON (user_id = id);
+ERROR:  relation local_table is not distributed
+-- CTEs should be able to terminate a router query
+WITH cte AS (
+	WITH cte_1 AS (
+		SELECT * FROM local_table WHERE id < 7
+	),
+	cte_2 AS (
+		SELECT * FROM local_table WHERE id > 3
+	),
+	cte_dist AS (
+		SELECT count(*) as u_id FROM users_table
+	),
+	cte_merge AS (
+		SELECT cte_1.id as id FROM cte_1 join cte_2 on TRUE
+	)
+	SELECT count(*) FROM users_table join cte_merge on id=user_id
+)
+SELECT
+  row_number() OVER (), count(*)
+FROM
+  cte, users_table
+WHERE
+  cte.count=user_id and user_id=5;
+ row_number | count 
+------------+-------
+          1 |     0
+(1 row)
+
+-- CTEs should be able to terminate a real-time query
+WITH cte AS (
+	WITH cte_1 AS (
+		SELECT * FROM local_table WHERE id < 7
+	),
+	cte_2 AS (
+		SELECT * FROM local_table WHERE id > 3
+	),
+	cte_dist AS (
+		SELECT count(*) as u_id FROM users_table
+	),
+	cte_merge AS (
+		SELECT cte_1.id as id FROM cte_1 join cte_2 on TRUE
+	)
+	SELECT count(*) FROM users_table join cte_merge on id=user_id
+)
+SELECT count(*) FROM cte, users_table where cte.count=user_id;
+ count 
+-------
+     0
+(1 row)
+
+SET citus.task_executor_type='task-tracker';
+-- CTEs shouldn't be able to terminate a task-tracker query
```

Comment:
Hmm, I expected to see the error `complex subqueries and CTEs are not supported when ...` , no?

Reply:
Good point, the final query should not do any re-partitioning, since that's not supported in combination with a subquery.